### PR TITLE
Update getResult method in RecordIteratorUtils class

### PR DESCRIPTION
**Pull request by lahirusamith**
## Purpose
$title

This change allows other db modules to override the setting and converting xml methods according to their db driver implementations. 

Fixes:

## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: native/src/main/java/io/ballerina/stdlib/sql/parameterprocessor/AbstractResultParameterProcessor.java [hunk 1]
```diff
@@ -215,6 +215,9 @@ public abstract Object processCustomOutParameters(
     public abstract Object processCustomTypeFromResultSet(ResultSet resultSet, int columnIndex,
                                                            ColumnDefinition columnDefinition) throws ApplicationError;
 
+    public abstract Object getAndConvertXml(ResultSet resultSet, int columnIndex, int sqlType, Type balType)
```
Comment (niveathika) [inline native/src/main/java/io/ballerina/stdlib/sql/parameterprocessor/AbstractResultParameterProcessor.java:218]:
```suggestion
    public Object processXmlResult(ResultSet resultSet, int columnIndex, int sqlType, Type balType) throws ApplicationError, SQLException {
        SQLXML sqlxml = resultSet.getSQLXML(columnIndex);
         return convertXml(sqlxml, sqlType, balType);
     }
     public abstract Object convertXml(sqlxml, sqlType, ballerinaType);
```

File: native/src/main/java/io/ballerina/stdlib/sql/parameterprocessor/DefaultResultParameterProcessor.java [hunk 2]
```diff
@@ -854,6 +854,13 @@ public Object processCustomTypeFromResultSet(ResultSet resultSet, int columnInde
         throw new ApplicationError("Unsupported SQL type " + columnDefinition.getSqlName());
     }
 
+    @Override
+    public Object getAndConvertXml(ResultSet resultSet, int columnIndex, int sqlType, Type balType)
```
Comment (niveathika) [inline native/src/main/java/io/ballerina/stdlib/sql/parameterprocessor/DefaultResultParameterProcessor.java:858]: With the changes I suggested, there should be no changes to module-specific ResultParameterProcessor.

File: native/src/main/java/io/ballerina/stdlib/sql/utils/RecordIteratorUtils.java [hunk 3]
```diff
@@ -181,8 +180,7 @@ private static Object getResult(ResultSet resultSet, int columnIndex, ColumnDefi
                 Struct structData = (Struct) resultSet.getObject(columnIndex);
                 return resultParameterProcessor.convertStruct(structData, sqlType, ballerinaType);
             case Types.SQLXML:
-                SQLXML sqlxml = resultSet.getSQLXML(columnIndex);
-                return resultParameterProcessor.convertXml(sqlxml, sqlType, ballerinaType);
+                return resultParameterProcessor.getAndConvertXml(resultSet, columnIndex, sqlType, ballerinaType);
```
Comment (niveathika) [inline native/src/main/java/io/ballerina/stdlib/sql/utils/RecordIteratorUtils.java:183]: Lets do the suggested changes for all data types in this switch statement.

File: native/src/main/java/io/ballerina/stdlib/sql/parameterprocessor/AbstractResultParameterProcessor.java (hunk 1 above)
Comment (lahirusamith) [reply native/src/main/java/io/ballerina/stdlib/sql/parameterprocessor/AbstractResultParameterProcessor.java]: updated in https://github.com/ballerina-platform/module-ballerina-sql/pull/329/commits/ec7082e053a0528f35ea6b01e858df0b0ba0fb6f

File: native/src/main/java/io/ballerina/stdlib/sql/parameterprocessor/DefaultResultParameterProcessor.java (hunk 2 above)
Comment (lahirusamith) [reply native/src/main/java/io/ballerina/stdlib/sql/parameterprocessor/DefaultResultParameterProcessor.java]: updated in https://github.com/ballerina-platform/module-ballerina-sql/pull/329/commits/ec7082e053a0528f35ea6b01e858df0b0ba0fb6f

File: native/src/main/java/io/ballerina/stdlib/sql/utils/RecordIteratorUtils.java (hunk 3 above)
Comment (lahirusamith) [reply native/src/main/java/io/ballerina/stdlib/sql/utils/RecordIteratorUtils.java]: updated in https://github.com/ballerina-platform/module-ballerina-sql/pull/329/commits/ec7082e053a0528f35ea6b01e858df0b0ba0fb6f

File: native/src/main/java/io/ballerina/stdlib/sql/parameterprocessor/AbstractResultParameterProcessor.java [hunk 4]
```diff
@@ -213,7 +225,141 @@ public abstract Object processCustomOutParameters(
             CallableStatement statement, int paramIndex, int sqlType) throws ApplicationError;
 
     public abstract Object processCustomTypeFromResultSet(ResultSet resultSet, int columnIndex,
-                                                           ColumnDefinition columnDefinition) throws ApplicationError;
+                                                           ColumnDefinition columnDefinition) throws ApplicationError,
+            SQLException;
+
+    public Object processArrayResult(ResultSet resultSet, int columnIndex, int sqlType, Type ballerinaType)
+            throws ApplicationError, SQLException {
+        return convertArray(resultSet.getArray(columnIndex), sqlType, ballerinaType);
```
Comment (niveathika) [inline native/src/main/java/io/ballerina/stdlib/sql/parameterprocessor/AbstractResultParameterProcessor.java:233]: Let's change this to 2 lines as before. Else it's harder to read.

Comment (lahirusamith) [reply native/src/main/java/io/ballerina/stdlib/sql/parameterprocessor/AbstractResultParameterProcessor.java]: done in https://github.com/ballerina-platform/module-ballerina-sql/pull/329/commits/9afd6cd860d444cb9e16b974ff74d0768c2cdb05

**Review by niveathika — APPROVED** [review]

**Pull request merged**